azure-functions-powershell-library/GetFunctionsMetadataCommand.cs

```diff
@@ -0,0 +1,48 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Common;
+using Microsoft.Azure.Functions.PowerShellWorker;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Functions.PowerShell
+{
+    [Cmdlet(VerbsCommon.Get, "FunctionsMetadata")]
+    public class GetFunctionsMetadataCommand : Cmdlet
+    {
+        /// <summary>
+        /// The path of the Azure Functions directory
+        /// </summary>
+        [Parameter(Mandatory = true, Position = 0)]
+        public string FunctionsAppDirectory { get; set; } = string.Empty;
+
+        private string outputJson { get; set; } = string.Empty;
+
+        protected override void ProcessRecord()
+        {
+            try
+            {
+                if (!string.IsNullOrEmpty(FunctionsAppDirectory) && Directory.Exists(FunctionsAppDirectory))
+                {
+                    List<FunctionInformation> bindingInformations = WorkerIndexingHelper.IndexFunctions(FunctionsAppDirectory);
+                    outputJson = System.Text.Json.JsonSerializer.Serialize(bindingInformations);
+                }
+                else
+                {
+                    ThrowTerminatingError(new ErrorRecord(new Exception("Functions app directory parameter is required and must be a valid directory"), "Invalid function app directory", ErrorCategory.ParserError, null));
```

Comment:
Customer facing script should go in a `.resx` file. Here is an example: https://github.com/Azure/azure-functions-powershell-worker/blob/dev/src/resources/PowerShellWorkerStrings.resx

Reply:
Done